### PR TITLE
Add git revision info to player version

### LINKED ISSUE #87: Use "git describe" to display a nice rev-number
After creating our first tag (0.1) "git describe" can be used to get a more readable revision number 0.1-### (where ### is the number of commits since 0.1).

The output of "git describe" should be written in some header file and then be included by player and e.g. displayed in the title bar.

This makes it easier for us to figure out which version somebody uses (Maybe also write it in the PNG-tEXT chunk under "Software" if he used our screenshot function and we cant see the title bar). And to detect if somebody made local changes (e.g. by adding an 'M' after the rev number)

### PR DESCRIPTION
**Pull request by carstene1ns**
Replaces #2070, fixes #87.

Works both on git checkouts and release tarballs.
A checked out git tag behaves the same like a release tarball.
`make distcheck` works

For our builds we can easily inject a tag (like CI, PR, AUR, OBS) with
`-DPLAYER_VERSION_APPEND` for cmake or `--enable-append-version` for autotools.

### CODE REVIEW

File: .github/workflows/stable-compilation.yml
Comment (fdelapena) [inline .github/workflows/stable-compilation.yml:55]: Looking good! Is there a simple way to override this date value later to support reproducible builds? Or an alternate, more deterministic way by getting the tag/commit date from git, for example (the release date could be different, however).

**Review by fdelapena — APPROVED** [review]

File: .github/workflows/stable-compilation.yml
Comment (carstene1ns) [reply .github/workflows/stable-compilation.yml:55]: Had a look at this before. Currently the date is just the default string if not provided. Maintainers can just overwrite it to empty for example. 

Some time in the future we can add support for `SOURCE_DATE_EPOCH`, which adds up on that. Then `PLAYER_DATE` could get introduced to implement this.

File: CMakeLists.txt
Comment (carstene1ns) [inline CMakeLists.txt:480]: CMake supports forcing timestamp since 3.8:
https://cmake.org/cmake/help/latest/command/string.html#timestamp

So this works ootb.

**Review by Ghabry — APPROVED** [review]

**Comment by carstene1ns** [comment]
Do not merge yet.

File: .github/workflows/stable-compilation.yml
Comment (carstene1ns) [reply .github/workflows/stable-compilation.yml:55]: Using an autoconf archive macro solves this for now.

File: CMakeLists.txt
Comment (carstene1ns) [inline CMakeLists.txt:1177]: This has been successive broken in 2016 and 2018 `m(`

doxygen used the version of the expat package here (because of config file providing `PACKAGE_VERSION`)

**Comment by carstene1ns** [comment]
test this please

File: CMakeLists.txt
Comment (carstene1ns) [inline CMakeLists.txt:954]: Something wrong here.
`PRODUCTVERSION 0,7` missing patch and tweak.

**Comment by carstene1ns** [comment]
This is ready now.

**Comment by Ghabry** [comment]
Not a full review due to lack of time.
Wow you really pushed this forward. Even updates all the version information in all of the resource files etc. Great job!

Also the oldest issue closed :tada:

**Pull request merged**
